bakery/src/scripts/link_single.py

```diff
@@ -166,18 +173,31 @@ def transform_links(
                 f"from link {link}"
             )
         patch_link(node, source_book_uuid, canonical_book_uuid,
-                   canonical_book_slug, page_slug)
+                   canonical_book_slug, page_slug, target_book)
 
     save_linked_collection(output_path, doc)
 
 
 def main():
-    (baked_content_dir,
+    (
+        baked_content_dir,
         baked_meta_dir,
         source_book_slug,
-        output_path) = sys.argv[1:5]
+        output_path,
+    ) = sys.argv[1:5]
+
+    if len(sys.argv) > 5:
```

Comment:
I believe even if `$TARGET_BOOK` is empty in `link_single.sh`, `sys.argv[5]` will exist, no? Or am I wrong / you found a need to do this check while testing?

Reply:
Since it was at the end, I think the shell script was dropping it.